src/assets/src/components/queueSettings.tsx

```diff
@@ -210,20 +213,29 @@ export function ManageQueueSettingsPage(props: PageProps<SettingsPageParams>) {
         setShowSuccessMessage(false);
     };
 
+    const handleLocationChange = (newQueueLocation: string) => {
+        setQueueLocation(newQueueLocation);
+        validateAndSetLocationResult(newQueueLocation);
+        setShowSuccessMessage(false);
+    };
+
     // On click handlers
     const handleSaveGeneralClick = () => {
         const curNameValidationResult = !nameValidationResult ? validateAndSetNameResult(name) : nameValidationResult;
         const curDescriptValidationResult = !descriptValidationResult
             ? validateAndSetDescriptResult(description) : descriptValidationResult;
         const curAllowedValidationResult = !allowedValidationResult
             ? validateAndSetAllowedResult(allowedMeetingTypes) : allowedValidationResult;
-        if (!curNameValidationResult.isInvalid && !curDescriptValidationResult.isInvalid && !curAllowedValidationResult.isInvalid) {
+        const curLocationValidationResult = !locationValidationResult 
+            ? validateAndSetLocationResult(queueLocation) : locationValidationResult;
+        if (!curNameValidationResult.isInvalid && !curDescriptValidationResult.isInvalid && !curAllowedValidationResult.isInvalid && !curLocationValidationResult) {
```

Comment:
Update is broken right now. Looks like you need to use `.isInvalid` on the location validation result you created. I'd also probably split up this long line for readability, like so:

```
        if (
            !curNameValidationResult.isInvalid &&
            !curDescriptValidationResult.isInvalid &&
            !curAllowedValidationResult.isInvalid &&
            !curLocationValidationResult.isInvalid
        ) {
```